woocommerce/compatibility/class-sv-wc-order-compatibility.php

```diff
@@ -522,6 +522,111 @@ public static function get_edit_order_url( \WC_Order $order ) {
 	}
 
 
+	/**
+	 * Gets the order meta according to HPOS availability.
+	 *
+	 * Uses WC_Order data store object if HPOS is enabled, otherwise it uses the traditional get_post_meta() method.
+	 *
+	 * @since x.y.z
+	 *
+	 * @param int $order_id Required. Order ID.
+	 * @param string $key Required. Meta key.
+	 * @param bool $single return first found meta with key, or all with $key. Defaults to true.
+	 *
+	 * @return mixed
+	 */
+	public static function get_order_meta( int $order_id, string $key, bool $single = true) {
+
+		$is_hpos_enabled = SV_WC_Plugin_Compatibility::is_hpos_enabled();
+		$value = false;
+
+		if ( $is_hpos_enabled ) {
+			if ( $order = wc_get_order( $order_id ) ) {
+				$value = $order->get_meta( $key, $single );
+			}
+		} else {
+			$value = get_post_meta( $order_id, $key, $single );
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Updates the order meta according to HPOS availability.
+	 *
+	 * Uses WC_Order data store object if HPOS is enabled, otherwise it uses the traditional update_post_meta() method.
+	 *
+	 * @since x.y.z
+	 *
+	 * @param int $order_id Required. Order ID.
+	 * @param string $key Required. Meta key.
+	 * @param mixed $value Required. Meta value.
+	 */
+	public static function update_order_meta( int $order_id, string $key, $value ) {
+
+		$is_hpos_enabled = SV_WC_Plugin_Compatibility::is_hpos_enabled();
+
+		if ( $is_hpos_enabled ) {
+			if ( $order = wc_get_order( $order_id ) ) {
+				$order->update_meta_data( $key, $value, $order_id );
```

Comment:
I believe we don't need to pass the order_id to `update_meta_data` - this is the meta id not the order ID; we could omit this argument entirely since it's not used with `update_post_meta()`

Reply:
@unfulvio-godaddy nice catch, fixed it.